realm/realm-library/src/main/java/io/realm/OrderedRealmCollectionSnapshot.java

```diff
@@ -0,0 +1,191 @@
+package io.realm;
+
+import io.realm.internal.Collection;
+import io.realm.internal.UncheckedRow;
+
+/**
+ * An {@link OrderedRealmCollectionSnapshot} is a special type of {@link OrderedRealmCollection}. It can be created by
+ * calling {@link OrderedRealmCollection#createSnapshot()}. Unlike {@link RealmResults} and {@link RealmList}, it's
+ * size and orders of elements will never be changed after creation.
```

Comment:
Should we mention something about invalid objects? And that `createSnapshot()` is almost idempotent?